### ISSUE TITLE
scripts: improving the -e option

### ISSUE THREAD

**Issue by nmav** [tomato42/tlsfuzzer#430]
# Feature request

As it is now the '-e' option allows to skip the result of a test. In order to expect fail a particular test one has to add two entries, one to skip the test originally, and another where the test whether `exp_pass` equals false. That is if one test fails from a script, the json should be:
```
         {"name" : "test-large-hello.py",
          "arguments" :
                  ["-e', "fragmented, padding ext 0 bytes"]},
         {"name" : "test-large-hello.py",
          "arguments" :
                  ["fragmented, padding ext 0 bytes"],
          "exp_pass" : false},
```
That's a very chatty way to write a test case, and becomes very convoluted, and hard to maintain when other command line parameters are needed (port, -n, etc).

A simpler approach would be to have a dedicated parameter to expect fail a particular test, e.g., `-x`. That would simplify the above json to a single entry:
```
         {"name" : "test-large-hello.py",
          "arguments" :
                  ["-x', "fragmented, padding ext 0 bytes"]},
```

**Comment by tomato42** [comment]
making it so that it also checks for a particular error message would also be nice...

**Comment by tomato42** [comment]
this will also require update to the `scripts_retention.py` to count the expected fail and error out on unexpected pass

**Comment by Stonemason5040** [comment]
This will have to be modified in each and every test. Do i get it right?

**Comment by Stonemason5040** [comment]
There are tests which don't have a `"-e"` option, should i add it as well as `"-x"` option

**Comment by tomato42** [comment]
yes, though I'd suggest to first figure out the mechanism to support this option, implement it in the `test-conversation.py` and `test-tls13-conversation.py` and only then start implementing it in the scripts themselves

**Comment by Stonemason5040** [comment]
Would you like to change the output to: 
`Test end`
`version: 2`
`total: 6`
`skip: 3`
`pass: 4`
`xfail: 1`
`fail: 1`
`xpass: 0`
and meaby add colours as well.

**Comment by tomato42** [comment]
`version` should not be in the same block as the test results, otherwise looks fine

**Comment by tomato42** [comment]
colours probably aren't a good idea: to do it portably we would need to use a library, and I want to keep dependencies to minimum

**Comment by Stonemason5040** [comment]
> colours probably aren't a good idea: to do it portably we would need to use a library, and I want to keep dependencies to minimum

i have found a way that works for all python versions and doesn't use any libraries.

**Comment by tomato42** [comment]
by portably I meant Windows, Linux and OSX

**Comment by tomato42** [comment]
fixed by #652
